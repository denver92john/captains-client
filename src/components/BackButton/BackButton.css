.backbutton-div {
    text-align: left;
}

.fa-backbutton-icon {
    font-size: 25px;
}

.backbutton-text {
    display: inline-block;
}