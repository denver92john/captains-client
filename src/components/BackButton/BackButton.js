import React from 'react';
import {Link} from 'react-router-dom';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';

function BackButton(props) {
    return (
        <div className="backbutton-div">
            <Link to={props.backTo}>
                <FontAwesomeIcon icon="arrow-circle-left" className="fa-backbutton-icon" />
                <p className="backbutton-text">{props.backWhat}</p>
            </Link>
        </div>
    );
}

export default BackButton;