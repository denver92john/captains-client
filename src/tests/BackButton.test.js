import React from 'react';
import ReactDOM from 'react-dom';
import {BrowserRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';
import BackButton from '../components/BackButton/BackButton';

describe('BackButton component', () => {
    it('renders without errors', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <BackButton backTo={'/dashboard'} backWhat={'Dashboard'} />
            </BrowserRouter>,
            div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders UI as expected', () => {
        const tree = renderer
            .create(<BrowserRouter>
                    <BackButton backTo={'/dashboard'} backWhat={'Dashboard'} />
                </BrowserRouter>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
})