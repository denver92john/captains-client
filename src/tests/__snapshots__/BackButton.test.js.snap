// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton component renders UI as expected 1`] = `
<div
  className="backbutton-div"
>
  <a
    href="/dashboard"
    onClick={[Function]}
  >
    <p
      className="backbutton-text"
    >
      Dashboard
    </p>
  </a>
</div>
`;
